src/Orleans/Streams/Internal/StreamConsumer.cs

```diff
@@ -172,6 +172,32 @@ public Task<StreamSubscriptionHandle<T>> SubscribeAsync(IAsyncObserver<T> observ
                                   .ToList<StreamSubscriptionHandle<T>>();
         }
 
+        public async Task Cleanup()
+        {
+            if (logger.IsVerbose) logger.Verbose("Cleanup() called");
+            if (myExtension == null)
+                return;
+
+            var allHandles = myExtension.GetAllStreamHandles<T>();
```

Comment:
I thought about it. Don't forget that this is per stream. So for each stream we will talk to its pubsub grain.
So you are suggesting to batch multiple subscriptions to the same stream and send them as one call to its pub sub grain.
This is where I thought it is a bit of a premature optimization, cause in majority of cases one would not create multiple subscriptions to the same stream. The down side is that will be quite a big code change, will nee to touch lots of files related to pub sub API and the grain, ...I did not want to complicate this PR even more. This was meant to be a relatively small, easy to review PR, just small cleanup.
PubSub API change makes it much bigger and complicated.
We can add this optimization later on, if you still think its worth it.